lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -382,11 +382,17 @@ sub _compose_job_overview_search_args {
 
     # determine build number
     if (!$search_args{build}) {
-        # yield the latest build of the first group if (multiple) groups but not build specified
-        # note: the search arg 'groupid' is ignored by complex_query() because we later assign 'groupids'
-        $search_args{groupid} = $groups[0]->id if (@groups);
-
-        $search_args{build} = $schema->resultset('Jobs')->latest_build(%search_args);
+        if (@groups){
+            my @builds;
+            @builds = map {
+                $search_args{groupid} = $_->{_column_data}->{id};
+                $schema->resultset('Jobs')->latest_build(%search_args);
+            } @groups;
+            $search_args{build} = \@builds;
```

Comment:
Some of the builds (or even all) gathered in the previous `map` might be undef.
```suggestion
            $search_args{build} = [grep { defined $_ } @builds];
```

Reply:
In my test I got `[ undef, ':TW.8318' ]` in `@builds`.

Yes, conditionally returning `()` would also work.

e.g.
```perl
$schema->resultset('Jobs')->latest_build(%search_args) || ();
```

One could even save another statement:
```perl
my @builds = map {
    $schema->resultset('Jobs')->latest_build(%search_args, groupid => $_->id) || ();
} @groups;
```

Reply:
To avoid duplicated build names (as you mentioned below) I suggest the following:
```perl
            my %builds;
            for my $group (@groups) {
                my $build = $schema->resultset('Jobs')->latest_build(%search_args, groupid => $group->id) or next;
                $builds{ $build }++;
            }
            $search_args{build} = [sort keys %builds];
```

Reply:
done